src/shogun/clustering/KMeans.cpp

```diff
@@ -77,35 +77,31 @@ void CKMeans::set_initial_centers(SGMatrix<float64_t> centers)
 
 void CKMeans::set_random_centers(SGVector<float64_t> weights_set, SGVector<int32_t> ClList, int32_t XSize)
 {
+	/*initialise the cluster centroids randomly among all data*/
 	CDenseFeatures<float64_t>* lhs=
 		CDenseFeatures<float64_t>::obtain_from_generic(distance->get_lhs());
 
-	for (int32_t i=0; i<XSize; i++)
+	SGVector<int32_t> Perm=SGVector<int32_t>(XSize);
```

Comment:
we usually name variables with lower cases.
Also I suggest to call it random_inds